txpool/tx_pool.go

```diff
@@ -235,6 +235,37 @@ func (p *TxPool) add(newTx *tx.Transaction, rejectNonExecutable bool, localSubmi
 		return nil
 	}
 
+	if err := p.checkOriginAndDelegatorBlocked(newTx); err != nil {
+		return err
+	}
+
+	if err := p.validateTxBasics(newTx); err != nil {
+		return err
+	}
+
+	txObj, err := ResolveTx(newTx, localSubmitted)
+	if err != nil {
+		return badTxError{err.Error()}
+	}
+
+	headSummary := p.repo.BestBlockSummary()
+	if isChainSynced(uint64(time.Now().Unix()), headSummary.Header.Timestamp()) {
+		err = p.addWhenSynced(newTx, txObj, headSummary, rejectNonExecutable, localSubmitted)
+	} else {
+		err = p.addWhenNotSynced(newTx, txObj)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	atomic.AddUint32(&p.addedAfterWash, 1)
+	metricTxPoolGauge().AddWithLabel(1, map[string]string{"source": source, "type": txTypeString})
+	return nil
+}
+
+// checkOriginAndDelegatorBlocked checks if the transaction origin or delegator is blocked.
+func (p *TxPool) checkOriginAndDelegatorBlocked(newTx *tx.Transaction) error {
```

Comment:
Should this return errors instead of nil now ?
Also likely ok to return error if the Origin() fails